src/hooks/useFocusTrap.ts

```diff
@@ -0,0 +1,46 @@
+// Based on https://hiddedevries.nl/en/blog/2017-01-29-using-javascript-to-trap-focus-in-an-element
+import { useRef, useEffect } from 'react';
+
+const KEYCODE_TAB = 9;
+
+export function useFocusTrap(visible = true) {
```

Comment:
const + type, e.g. export const useFocusTrap = (visible: boolean = true): **whatever return type elref is goes here** => {